flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModel.java

```diff
@@ -0,0 +1,220 @@
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {model};
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        DataStream<Row> data = tEnv.toDataStream(inputs[0]);
+        final String broadcastModelName = "broadcastModel";
+        DataStream<LogisticRegressionModelData> modelData =
+                tEnv.toDataStream(model).map(x -> (LogisticRegressionModelData) x.getField(0));
+        DataStream<Row> predictResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(data),
+                        Collections.singletonMap(broadcastModelName, modelData),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.transform(
+                                    "predict",
+                                    new RowTypeInfo(
+                                            new TypeInformation[] {
+                                                PrimitiveArrayTypeInfo
+                                                        .DOUBLE_PRIMITIVE_ARRAY_TYPE_INFO,
+                                                BasicTypeInfo.DOUBLE_TYPE_INFO,
+                                                PrimitiveArrayTypeInfo
+                                                        .DOUBLE_PRIMITIVE_ARRAY_TYPE_INFO
+                                            },
+                                            new String[] {
+                                                getFeaturesCol(),
+                                                getPredictionCol(),
+                                                getPredictionDetailCol()
+                                            }),
+                                    new PredictOp(
+                                            new PredictOneRecord(
+                                                    getFeaturesCol(), broadcastModelName)));
+                        });
+
+        return new Table[] {tEnv.fromDataStream(predictResult)};
+    }
+
+    /** A utility operator used for prediction. */
+    private static class PredictOp extends AbstractUdfStreamOperator<Row, RichMapFunction<Row, Row>>
+            implements OneInputStreamOperator<Row, Row> {
+        public PredictOp(RichMapFunction<Row, Row> userFunction) {
+            super(userFunction);
+        }
+
+        @Override
+        public void processElement(StreamRecord<Row> streamRecord) throws Exception {
+            output.collect(new StreamRecord<>(userFunction.map(streamRecord.getValue())));
+        }
+    }
+
+    /** A utility function used to predict one input record. */
+    private static class PredictOneRecord extends RichMapFunction<Row, Row> {
+
+        String broadcastModelName;
+
```

Comment:
It might be better to reduce the empty lines here.

Reply:
As a common practice in Flink, we usually left one empty line before/after each class variable.